src/StreamUtils.js

```diff
@@ -0,0 +1,163 @@
+("use strict");
+
+// System dependencies (Built in modules)
+const crypto = require("crypto");
+const { Transform, Writable } = require("stream");
+
+// Project dependencies
+const Utils = require("./Utils");
+
+/**
+ * A transform stream to prepare files for web serving by setting headers and configuring slug extenions
+ * @class StreamUtils
+ */
+class StreamUtils {
+  /**
+   *Creates an instance of StreamUtils.
+   * @memberof StreamUtils
+   */
+  constructor() {
+    // Set up object/arrays to track files of various states
+    this.destinationReads = {};
+    this.destinationDeletes = [];
+    this.destinationUpdates = [];
+    this.destinationCreates = [];
+  }
+
+  /**
+   * Get the MD5 hash of the fileContents. Note that we opted for MD5 as that is already in use by S3.
+   * @static
+   * @param {ArrayBuffer} fileContents: Typically found as .contents from a ReadableStream where objectMode = true;
+   * @returns:                          A MD5 hash of the file contents
+   * @memberof Utils
+   */
+  _getFileHash(fileContents) {
+    const shasum = crypto.createHash("md5");
+    shasum.update(fileContents);
+    return shasum.digest("hex");
+  }
+
+  /**
+   * Helper function to set various additional properties on a VinylFile
+   *
+   * @static
+   * @param {*} vinylFile
+   * @param {*} options
+   * @returns
+   * @memberof Metafy
```

Comment:
Which class is Metafy?  I see it referenced in the comments a few times.

Reply:
Hmm mobile interface did not give me this option to reply (now on desktop).

Metafy was a poor name that I used as a placeholder with the intent of updating. Which I did, to StreamUtils, which is only marginally better.

I have been sketching out a revamped layout and naming convention that I will ticket into Boards soon.